distributed_shampoo/utils/shampoo_preconditioner_list.py

```diff
@@ -315,17 +316,22 @@ class ShampooKroneckerFactorsState(OptimizerModule):
     """Shampoo Kronecker Factors (wrapped) for storing in the optimizer state."""
 
     factor_matrices: Tuple[QuantizedTensor, ...]
-    inv_factor_matrices: Tuple[QuantizedTensor, ...]
     factor_matrix_indices: Tuple[str, ...]
     is_factor_matrices_diagonal: Tuple[Tensor, ...] = field(init=False)
+    inv_factor_matrices: Optional[Tuple[QuantizedTensor, ...]] = None
+    factor_matrices_eigenvectors: Optional[Tuple[QuantizedTensor, ...]] = None
```

Comment:
This makes sense in terms of allowing us to share the code path below. I kept them separate because I couldn't think of a semantically meaningful name for both of them, so I thought it would make the code more readable to have two distinct names.

One general question on simplifying the code path: I have also considered just adding a new class, say `EigenvalueCorrectedShampooPreconditionerList`, that inherits from `ShampooPreconditionerList`. In this case, we could avoid the if-else statements in the methods of `ShampooPreconditionerList` that check if `use_eigenvalue_correction == True`. I haven't thought this fully through, but do you think this would be cleaner?